### PR TITLE
Build SPIRV-Tools as a shared library

### LINKED ISSUE #1240: Linking SPIRV-Tools with C code (shared/dynamic library request)
Using SPIRV-Tools in C code is a bit problematic.

Let's take as an example the following code:
```
#include <stdio.h>
#include "spirv-tools/libspirv.h"

int main(int argc, char **argv)
{
    spv_context context = spvContextCreate(SPV_ENV_VULKAN_1_0);
    spvContextDestroy(context);
    return 0;
}
```

The usual way of building it fails:
```
$ gcc main.c $(pkg-config --cflags --libs SPIRV-Tools)

/home/joseph/src/vulkan/install/lib64/libSPIRV-Tools.a(table.cpp.o): In function `spvContextCreate':
/home/joseph/src/vulkan/SPIRV-Tools/source/table.cpp:52: undefined reference to `operator new(unsigned long)'
/home/joseph/src/vulkan/install/lib64/libSPIRV-Tools.a(table.cpp.o): In function `spvContextDestroy':
/home/joseph/src/vulkan/SPIRV-Tools/source/table.cpp:55: undefined reference to `operator delete(void*)'

...

collect2: error: ld returned 1 exit status
```

I can fix this by linking with `libstdc++` or by using `g++`, but I would like to avoid this if possible.

Could SPIRV-Tools be also built as a shared library?

### PR DESCRIPTION
**Pull request by jozefkucia**
Fixes #1240.

Tested on Linux.

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -156,7 +159,8 @@ add_spvtools_unittest(
 add_spvtools_unittest(
   TARGET c_interface_shared
   SRCS c_interface_test.cpp
-  LIBS ${SPIRV_TOOLS}-shared)
+  LIBS ${SPIRV_TOOLS}-shared
+  ENVIRONMENT PATH=$<TARGET_FILE_DIR:${SPIRV_TOOLS}-shared>)
```
Comment (dneto0) [inline test/CMakeLists.txt]: This is needed Windows so it can find the DLL.
I wonder why Linux (and OSX) don't need a modification for LD_LIBRARY_PATH?

**Comment by dneto0** [comment]
Rebased, squashed, and pushed into master as 3013897556

File: test/CMakeLists.txt (hunk 1 above)
Comment (jozefkucia) [reply test/CMakeLists.txt]: I think the answer is `RUNPATH`:

```
$ objdump -x test_c_interface_shared | grep PATH
  RUNPATH              /home/joseph/src/vulkan/SPIRV-Tools/build/source
```

**Comment by jozefkucia** [comment]
Thanks!

**Pull request closed without merging**